examples/hackernews/back-end/reactive_service/src/hackernews.service.ts

```diff
@@ -26,38 +25,33 @@ type Upvote = {
 
 type Upvoted = Post & { upvotes: number; author: User };
 
-export default class HackerNewsService implements SkipService {
-  initialData: {
-    posts: Entry<string, Post>[];
-    users: Entry<string, User>[];
-    upvotes: Entry<string, Upvote>[];
-  };
-  resources = { posts: PostsResource };
-
-  constructor(
-    posts: Entry<string, Post>[],
-    users: Entry<string, User>[],
-    upvotes: Entry<string, Upvote>[],
-  ) {
-    this.initialData = { posts, users, upvotes };
-  }
-
-  createGraph(inputCollections: {
-    posts: EagerCollection<number, Post>;
-    users: EagerCollection<number, User>;
-    upvotes: EagerCollection<number, Upvote>;
-  }): Record<string, EagerCollection<Json, Json>> {
-    const upvotes = inputCollections.upvotes.map(UpvotesMapper);
-    const postsWithUpvotes = inputCollections.posts.map(
-      PostsMapper,
-      inputCollections.users,
-      upvotes,
-    );
+type Inputs = {
+  posts: EagerCollection<number, Post>;
+  users: EagerCollection<number, User>;
+  upvotes: EagerCollection<number, Upvote>;
+};
+type Outputs = {
+  postsWithUpvotes: EagerCollection<number, Upvoted>;
+};
 
-    return {
-      postsWithUpvotes,
-    };
-  }
+export function serviceWithInitialData(
+  posts: Entry<string, Post>[],
+  users: Entry<string, User>[],
+  upvotes: Entry<string, Upvote>[],
+): SkipService {
+  return {
+    initialData: { posts, users, upvotes },
+    resources: { posts: PostsResource },
+    createGraph: (inputCollections: Inputs) => {
```

Comment:
I think the return type annotation is important to make the connection with the argument of `instantiate`
```suggestion
    createGraph: (inputCollections: Inputs): Outputs => {
```